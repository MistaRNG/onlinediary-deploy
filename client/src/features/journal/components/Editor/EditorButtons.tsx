import { Button } from "@blueprintjs/core";

interface ListItem {
  style: any;
  icon: any;
}

interface EditorButtonsProps {
  mouseDownHandler: (style: string, isList?: boolean) => (event: React.MouseEvent<HTMLButtonElement>) => void;
  styles: any[];
  lists: ListItem[];
}

const EditorButtons: React.FC<EditorButtonsProps> = ({ mouseDownHandler, styles, lists }) => {
  const StyleButtons = styles.map((style) => (
    <Button
      key={style}
      icon={style}
      minimal={true}
      onMouseDown={mouseDownHandler(style)}
    />
  ));

  const ListButtons = lists.map(({ style, icon }) => (
    <Button
      key={style}
      minimal={true}
      icon={icon}
      onMouseDown={mouseDownHandler(style, true)}
    />
  ));

  return (
    <div className="Editor-Buttons">
      {StyleButtons}
      {ListButtons}
    </div>
  );
};

export default EditorButtons;
