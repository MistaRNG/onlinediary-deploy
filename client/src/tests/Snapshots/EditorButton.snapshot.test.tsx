import React from 'react';
import { render } from '@testing-library/react';
import EditorButtons from '../../features/journal/components/Editor/EditorButtons';
import '@testing-library/jest-dom';
import { jest, describe, it, expect } from '@jest/globals';

describe('EditorButtons Component Snapshot Tests', () => {
  it('should render the EditorButtons component correctly', () => {
    // ARRANGE
    const mockMouseDownHandler = (style: string, isList?: boolean) => jest.fn((event: React.MouseEvent<HTMLButtonElement>) => {});

    const mockStyles = ['bold', 'italic', 'underline'];
    const mockLists = [
      { style: 'unordered-list-item', icon: 'list' },
      { style: 'ordered-list-item', icon: 'numbered-list' },
    ];

    // ACT
    const { asFragment } = render(
      <EditorButtons
        mouseDownHandler={mockMouseDownHandler}
        styles={mockStyles}
        lists={mockLists}
      />
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
