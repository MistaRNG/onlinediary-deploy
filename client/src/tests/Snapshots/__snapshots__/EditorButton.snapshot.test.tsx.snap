// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorButtons Component Snapshot Tests should render the EditorButtons component correctly 1`] = `
<DocumentFragment>
  <div
    class="Editor-Buttons"
  >
    <button
      class="bp4-button bp4-minimal"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-bold"
        icon="bold"
      >
        <svg
          data-icon="bold"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M11.7 7c.2-.4.3-1 .3-1.5v-.4V5c0-.1 0-.2-.1-.3v-.1C11.4 3.1 10.1 2 8.5 2H4c-.5 0-1 .4-1 1v10c0 .5.4 1 1 1h5c2.2 0 4-1.8 4-4 0-1.2-.5-2.3-1.3-3zM6 5h2c.6 0 1 .4 1 1s-.4 1-1 1H6V5zm3 6H6V9h3c.6 0 1 .4 1 1s-.4 1-1 1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
    <button
      class="bp4-button bp4-minimal"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-italic"
        icon="italic"
      >
        <svg
          data-icon="italic"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M9.8 4H11c.5 0 1-.4 1-1s-.4-1-1-1H7c-.5 0-1 .4-1 1s.4 1 1 1h.8l-1.6 8H5c-.5 0-1 .4-1 1s.4 1 1 1h4c.5 0 1-.4 1-1s-.4-1-1-1h-.8l1.6-8z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
    <button
      class="bp4-button bp4-minimal"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-underline"
        icon="underline"
      >
        <svg
          data-icon="underline"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 14c2.8 0 5-2.2 5-5V3c0-.6-.4-1-1-1s-1 .4-1 1v6c0 1.7-1.3 3-3 3s-3-1.3-3-3V3c0-.6-.4-1-1-1s-1 .4-1 1v6c0 2.8 2.2 5 5 5zM13.5 15h-11c-.3 0-.5.2-.5.5s.2.5.5.5h11c.3 0 .5-.2.5-.5s-.2-.5-.5-.5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
    <button
      class="bp4-button bp4-minimal"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-list"
        icon="list"
      >
        <svg
          data-icon="list"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
    <button
      class="bp4-button bp4-minimal"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-numbered-list"
        icon="numbered-list"
      >
        <svg
          data-icon="numbered-list"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M2.76 7h1.26V0h-.94c-.04.21-.12.39-.25.54-.13.15-.29.27-.48.36-.18.09-.39.16-.62.2-.23.04-.46.06-.71.06v.9h1.74V7zm-.59 7.17c.18-.12.37-.25.58-.37a10.763 10.763 0 001.24-.83c.2-.16.37-.33.52-.51.15-.19.28-.39.37-.61.09-.22.14-.47.14-.74 0-.22-.04-.45-.12-.7-.08-.26-.21-.49-.4-.69-.18-.21-.43-.39-.72-.52-.3-.14-.68-.21-1.12-.21-.41 0-.77.07-1.08.2-.32.14-.58.32-.8.56-.22.23-.38.51-.49.84-.11.32-.16.67-.16 1.05h1.19c.01-.24.03-.47.08-.67.05-.21.11-.39.21-.54.09-.15.22-.27.38-.36.16-.09.35-.13.59-.13.26 0 .47.04.63.12.16.08.29.18.38.3.09.12.15.25.18.39s.05.27.05.4c-.01.27-.08.5-.22.71-.14.21-.32.4-.53.57-.22.18-.45.34-.71.49-.26.15-.51.31-.74.47-.5.31-.89.68-1.17 1.11-.3.41-.44.91-.45 1.48h5v-1H1.43c.05-.15.14-.29.27-.43.14-.13.29-.26.47-.38zM15.01 1.99h-7c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1h7c.55 0 1-.45 1-1v-1c0-.55-.44-1-1-1zm0 9h-7c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1h7c.55 0 1-.45 1-1v-1c0-.55-.44-1-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
